modules/ROOT/pages/apikit-4-scaffolding-reference.adoc

```diff
@@ -0,0 +1,239 @@
+= Scaffolding Reference
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:page-aliases: creating-an-apikit-4-project-with-maven.adoc
+
+Generate the backbone of a REST API implementation based on RESTful API Modeling Language (RAML) or OpenAPI Specifications (OAS) with APIkit for REST scaffolder.
+
+== Tooling
+
+Studio offers several ways to get an API scaffolded:
+
+* From *New Mule Project wizard*, select *Scaffold Flows From These API Specifications* when importing a published API from Exchange or Maven, when importing from a local file, or when downloading from Design Center.
+* From *Package Explorer*, right-click the API file and select *Generate flows from Local REST API* from the *Mule* context menu.
+* From *Package Explorer*, right-click the API dependency and select *Generate flows* from its context menu.
+
+NOTE: Scaffolding from a local file is not supported for OAS. Scaffolding OAS APIs is available only for published assets in Exchange.
+
+== Implementation Reference
+
+When scaffolding a REST API specification, APIkit generates the following components:
+
+* A *main flow* that act as the entry point and includes:
+** *HTTP Listener* as a source
+** *APIkit Router* as the main operation
++
+The *APIkit Router* is a crucial message processor that directs messages from incoming API requests to the corresponding Mule flows and generates a serialized response. It enhances messages by including default values. For instance, if a request is invalid, APIkit produces a `bad request` response. If the requested API resource lacks an associated flow, APIkit returns a `not implemented` response.
```

Comment:
What does "enhances messages by including default values" mean? Is bad request a default value?

Reply:
Yes! You're right! "Bad request" is a default value for queries that do not comply with the standard.
Thanks!

Reply:
OK, it's not that clear to me.

Reply:
@Crispy-Salesforce

Reply:
Hi, @hannanelson!
Is it clearer now? Thanks!

Reply:
I don't see the difference. I guess I'd like to see a connection between the default values and the different responses. See suggestion.

Reply:
OK! Thanks!